### PR TITLE
[DOCS-1776] Synthetics ICMP Tests

### PR DESCRIPTION
**Pull request by rmshree**
### What does this PR do?

Add documentation page for ICMP tests.

### Motivation
New feature release: https://docs.google.com/document/d/1krqrN0vxeVDXwYqZHe1afTBHTojZuSRajELMgFfToV4/edit#heading=h.svwxvwkd5dm2

### Preview

https://docs-staging.datadoghq.com/mahashree.rajendran/icmp_test/synthetics/api_tests/icmp_tests/

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/synthetics/api_tests/_index.md [hunk 1]
```diff
@@ -40,14 +43,15 @@ API tests are useful to help you **proactively monitor that your most important
 - [`SSL` test][2]
 - [`TCP` test][3]
 - [`DNS` test][4]
+- [`ICMP` test][5]
 
 {{< img src="synthetics/api_tests/api_tests.mp4" alt="API Test subtypes" video="true"  width="100%" >}}
```
Comment (MargotLepizzera) [inline content/en/synthetics/api_tests/_index.md:48]: We'll need a new video here :)

File: content/en/synthetics/api_tests/icmp_tests.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  text: "Get started with API tests"
+- link: "/synthetics/private_locations"
+  tag: "Documentation"
+  text: "Run ICMP Pings on internal endpoints"
+---
+
+
+## Overview
+
+ICMP tests allow you to easily **monitor the availability of your hosts and diagnose network communication issues to verify responsiveness from your application's endpoints**. By asserting on the values received from one or more **ICMP ping(s)** to your endpoint, Datadog can help **detect availability/connectivity issues, above-quota latency or hops, and changes in security firewall configuration**. ICMP tests can run from [managed][1] and [private locations][2] depending on whether you are willing to trigger ICMP pings to your endpoints from outside or inside your network.
```
Comment (MargotLepizzera) [inline content/en/synthetics/api_tests/icmp_tests.md:23]:
```suggestion
ICMP tests allow you to easily **monitor the availability of your hosts** and **diagnose network communication issues**. By asserting on the values received from one or more **ICMP ping(s)** to your endpoint, Datadog can help **detect connectivity issues, above-quota latency or hops, and unexpected changes in security firewall configuration**.  
 
ICMP tests can run from [**managed**][1] and [**private locations**][2] depending on whether you are willing to trigger ICMP pings to your endpoints from outside or inside your network.
```

Can you also add the bit about the type of runs? (ie. scheduled, on demand, etc.).

File: content/en/synthetics/api_tests/icmp_tests.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+## Configuration
+
+After choosing the type of test you are willing to create ([`HTTP`][3], [`SSL`][4], [`TCP`][5], [`DNS`][6], or [`ICMP` test][7]), you can define your test's request.
+
+### Define Request
+
+![][8]
```
Comment (MargotLepizzera) [inline content/en/synthetics/api_tests/icmp_tests.md:31]: Fix screenshot

File: content/en/synthetics/api_tests/icmp_tests.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+1. Specify the **domain name** or **IP address** to run your test on.
```
Comment (MargotLepizzera) [inline content/en/synthetics/api_tests/icmp_tests.md:33]:
```suggestion
1. Specify the **Domain Name** or **IP address** to run your test on.
```

File: content/en/synthetics/api_tests/icmp_tests.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+2. Select the **number of pings** to trigger per test session. By default, the number of pings is set to 4 - you can choose to increase or decrease.
```
Comment (MargotLepizzera) [inline content/en/synthetics/api_tests/icmp_tests.md:34]:
```suggestion
2. Select the **Number of Pings** to trigger per test session. By default, the number of pings is set to 4 - you can choose to increase or decrease.
```

File: content/en/synthetics/api_tests/icmp_tests.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+3. **Name** your ICMP test.
+4. Add env **Tags** as well as any other tag to your ICMP test. You can then use these tags to quickly filter through your Synthetic tests on the [Synthetic Monitoring homepage][9].
```
Comment (MargotLepizzera) [inline content/en/synthetics/api_tests/icmp_tests.md:36]:
```suggestion
4. Add `env` **Tags** as well as any other tag to your ICMP test. You can then use these tags to quickly filter through your Synthetic tests on the [Synthetic Monitoring homepage][9].
```

File: content/en/synthetics/api_tests/icmp_tests.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+5. Select the **Locations** to run your ICMP test from: ICMP tests can run from [managed][1] and [private locations][2] depending on whether you are willing to trigger the ICMP pings from outside or inside your network.
+
+Click on **Test URL** to try out the request configuration. You will see a response preview show up on the right side of your screen.
+
+### Specify test frequency
+
+ICMP tests can run:
+
+* **On a schedule** to ensure your most important services are always accessible to your users. Select the frequency you want Datadog to run your ICMP test.
```
Comment (MargotLepizzera) [inline content/en/synthetics/api_tests/icmp_tests.md:45]: Add the screenshot that is used in other tests?

File: content/en/synthetics/api_tests/icmp_tests.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+
+* [**Within your CI/CD pipelines**][10].
+* **On-demand** to run your tests whenever makes the most sense for your teams.
+
+
+### Define Assertions
+
+Assertions define what an expected test result is. When hitting `Test URL` basic assertions on `latency`, `packet loss` and ICMP ping result is added. You must define at least one assertion for your test to monitor.
```
Comment (MargotLepizzera) [inline content/en/synthetics/api_tests/icmp_tests.md:53]:
```suggestion
Assertions define what an expected test result is. When hitting `Test URL` basic assertions on `latency`, `packet loss` and `ICMP ping result` is added. You must define at least one assertion for your test to monitor.
```

File: content/en/synthetics/api_tests/icmp_tests.md [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+
+| Type          | Operator                                                                               | Value Type       |
+|---------------|----------------------------------------------------------------------------------------|------------------|
+| latency       | `is less than`, `is less than or equal`, `is`, `is more than`, `is more than or equal` | _integer (ms)_   |
```
Comment (MargotLepizzera) [inline content/en/synthetics/api_tests/icmp_tests.md:57]: This table is lacking the aggregation method bit for the latency assertions

File: content/en/synthetics/api_tests/icmp_tests.md [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+| packet loss   | `is less than`, `is less than or equal`, `is`, `is more than`, `is more than or equal` | _percentage (%)_ |
+| ICMP ping     | `Successful` or `Failed`                                                               | _string_         |
+
+You can create up to 10 assertions per API test by clicking on **New Assertion** or by clicking directly on the response preview:
```
Comment (MargotLepizzera) [inline content/en/synthetics/api_tests/icmp_tests.md:61]: Add screenshot

**Comment by MargotLepizzera** [comment]
@rmshree 

A few notes in addition to the inlines ones:

- You'll also need to update https://docs-staging.datadoghq.com/mahashree.rajendran/icmp_test/synthetics/ to include ICMP tests :) 
- Good call updating all other test subtypes!
- We're missing the reference to ICMP test in the menu, see https://a.cl.ly/RBu9Dj6m. IIRC the file that needs to be updated is https://github.com/DataDog/documentation/blob/master/config/_default/menus/menus.en.yaml

File: content/en/synthetics/api_tests/icmp_tests.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+| `INVALID_REQUEST` | The configuration of the test is invalid (for example, a typo in the URL).                                                                                                                                                                                                                                                                                                                                                                                     |
+| `SSL`             | The SSL connection couldn't be performed. [See the dedicated error page for more information][8].                                                                                                                                                                                                                                                                                                                                                      |
```
Comment (MargotLepizzera) [inline content/en/synthetics/api_tests/icmp_tests.md:143]: Let's completely get rid of the SSL line - I recently removed it on DNS & TCP tests :)

File: content/en/synthetics/api_tests/icmp_tests.md [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+
+{{< img src="synthetics/api_tests/notifications-example.png" alt="API Test Notifications"  style="width:90%;" >}}
+
+Click on **Save** to save your test and have Datadog start executing it.
+
+## Variables
+
+### Create local variables
+
+You can create local variables by defining their values from one of the below available builtins:
```
Comment (jtappa) [inline content/en/synthetics/api_tests/icmp_tests.md:118]:
```suggestion
You can create local variables by defining their values from one of the below available built-ins:
```

File: content/en/synthetics/api_tests/icmp_tests.md [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+### Use variables
+
+You can use the [global variables defined in the `Settings`][13] and the [locally defined variables](#create-local-variables) in the URL and assertions of your ICMP tests.
+To display your list of variables, type `{{` in your desired field:
+
+{{< img src="synthetics/api_tests/use_variable.mp4" alt="Using Variables in API tests" video="true" width="90%" >}}
+
+## Test failure
+
+A test is considered `FAILED` if it does not satisfy one or several assertions or if the request prematurely failed. In some cases, the test can indeed fail without being able to test the assertions against the endpoint, these reasons include:
```
Comment (jtappa) [inline content/en/synthetics/api_tests/icmp_tests.md:137]:
```suggestion
A test is considered `FAILED` if it does not satisfy one or several assertions or if the request prematurely failed. In some cases, the test can fail without being able to test the assertions against the endpoint, these reasons include:
```

**Review by jtappa — CHANGES_REQUESTED** [review]

**Review by jtappa — APPROVED** [review]

**Pull request merged**